Sources/TUSKit/TUSClient.swift

```diff
@@ -0,0 +1,523 @@
+            return id
+        } catch let error as TUSClientError {
+            throw error
+        } catch let error {
+            throw TUSClientError.couldNotStoreFile(underlyingError: error)
+        }
+    }
+    
+    // MARK: - Upload multiple files
+    
+    /// Upload multiple files by giving their url.
+    /// If you want a different uploadURL for each file, then please use `uploadFileAt(:)` individually.
+    /// - Parameters:
+    ///   - filePaths: An array of filepaths, represented by URLs
+    ///   - uploadURL: The URL to upload to. Leave nil for the default URL.
+    ///   - customHeaders: Any headers you want to add to the upload
+    ///   - context: Add a custom context when uploading files that you will receive back in a later stage. Useful for custom metadata you want to associate with the upload. Don't put sensitive information in here! Since a context will be stored to the disk.
+    /// - Returns: An array of ids
+    /// - Throws: TUSClientError
+    @discardableResult
+    public func uploadFiles(filePaths: [URL], uploadURL:URL? = nil, customHeaders: [String: String] = [:], context: [String: String]? = nil) throws -> [UUID] {
+        try filePaths.map { filePath in
+            try uploadFileAt(filePath: filePath, uploadURL: uploadURL, customHeaders: customHeaders, context: context)
+        }
+    }
+    
+    /// Upload multiple files by giving their url.
+    /// If you want a different uploadURL for each file, then please use `upload(data:)` individually.
+    /// - Parameters:
+    ///   - dataFiles: An array of data to be uploaded.
+    ///   - preferredFileExtension: A file extension to add when saving the file. E.g. You can add ".JPG" to raw data that's being saved. This will help the uploader's metadata.
+    ///   - uploadURL: The URL to upload to. Leave nil for the default URL.
+    ///   - customHeaders: Any headers you want to add to the upload
+    ///   - context: Add a custom context when uploading files that you will receive back in a later stage. Useful for custom metadata you want to associate with the upload. Don't put sensitive information in here! Since a context will be stored to the disk.
+    /// - Returns: An array of ids
+    /// - Throws: TUSClientError
+    @discardableResult
+    public func uploadMultiple(dataFiles: [Data], preferredFileExtension: String? = nil, uploadURL: URL? = nil, customHeaders: [String: String] = [:], context: [String: String]? = nil) throws -> [UUID] {
+        try dataFiles.map { data in
+            try upload(data: data, preferredFileExtension: preferredFileExtension, uploadURL: uploadURL, customHeaders: customHeaders, context: context)
+        }
+    }
+    
+    // MARK: - Cache
+    
+    /// Throw away all files.
+    /// - Important:This will clear the storage directory that you supplied.
+    /// - Important:Don't call this while the client is active. Only between uploading sessions.
```

Comment:
I added documentation on how to check (you can use the `remainingUploads`) property.
Yes you will get errors. I could make it more foolproof by prohibiting this from working, but I rather give implementers full control of the situation, even if that means errors occur.

E.g. Let's say you're logging out and want an instant delete of everything, that should be allowed, even if the TUSClient uploader starts to throw errors that it can't find files anymore to upload. 
Alternatively, we can say "You must stop the client before deleting", but I think that would just move the complexity to a different location. Thoughts?

Reply:
I agree lets keep it as is.